### PR TITLE
TestItem framework

### PR DESCRIPTION
**Pull request by gdalle**
Replace the usual tests with the new [testitem framework](https://discourse.julialang.org/t/prerelease-of-new-testing-framework-and-test-run-ui-in-vs-code/86355) for VSCode. This allows tests to be run independently and in parallel.

Solves #46 

By the way, I think we can still see the plots: there is a button "Show output" at the top of the testing tab

### CODE REVIEW

**Comment by gdalle** [comment]
@BatyLeo the tests of your general imitation loss are commented out

**Review by BatyLeo — APPROVED** [review]
Great changes !
I'll update the imitation loss tests to the new setup before this is merged

File: test/InferOptTestUtils/pipeline.jl [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            encoder,
+            pipeline_loss,
+            error_function,
+            cost,
+        )
+        Flux.train!(pipeline_loss, Flux.params(encoder), zip(data_train...), opt)
+    end
+
+    test_perf(perf_storage; decrease=decrease)
+    plot_perf(perf_storage)
```
[the remark this answers is not in the document]
Comment (gdalle) [reply test/InferOptTestUtils/pipeline.jl:39]: Good point. And btw the results can indeed be seen in VSCode, in the panel you only need to open the test outputs

[the remark this answers is not in the document]
Comment (gdalle) [reply test/InferOptTestUtils/pipeline.jl:39]:
```suggestion
    print_plot_perf(perf_storage)
```

**Comment by BatyLeo** [comment]
@gdalle `ImitationLoss` tests are updated

Should we remove/disable nightly from CI (and add 1.8) before merging ?

**Comment by gdalle** [comment]
> Should we remove/disable nightly from CI (and add 1.8) before merging ?

What I usually see in CI setups is that people test on

- the earliest version supported (here 1.7, but we could easily go down to 1.6)
- the latest release (called 1, since yesterday it's 1.9)
- nightly

I don't think we need to cover all versions in between. As for the bug on nightly, it's getting fixed soon: https://github.com/FluxML/Zygote.jl/pull/1420

The better option would be to ignore nightly for the badge. I opened a topic here (https://discourse.julialang.org/t/ignoring-nightly-failure-for-ci-badge/98028) but the solution I adopted doesn't seem to work anymore so :shrug:

**Pull request merged**